api/v1alpha7/conversion.go

```diff
@@ -509,6 +559,12 @@ func Convert_v1alpha7_OpenStackClusterSpec_To_v1alpha8_OpenStackClusterSpec(in *
 		}
 	}
 
+	if in.ManagedSecurityGroups {
+		out.ManagedSecurityGroups = &infrav1.SecurityGroupsSpec{
+			AllNodesSecurityGroupRules: legacyCalicoSecurityGroupRules,
+		}
```

Comment:
This isn't correct, because there's an interaction with `allowAllInClusterTraffic`. See https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/1878

We should probably unit test this conversion.